### PR TITLE
Feat/merge eval connect ast

### LINKED ISSUE #699: TODO: investigate merging evaluate() with connect() with the help of flags

### PR DESCRIPTION
**Pull request by bigopon**
# Pull Request

## 📖 Description

Change AST `.evaluate()` so that it support **optional** connect during the evaluation. This should help increase perf of our binding, as it can avoid doing double work of iterating the ast and observing after evaluating.

### 🎫 Issues

Resolves #699 

## 👩‍💻 Reviewer Notes

Interpolation binding isn't  touched, as the way it works at the moment requires some focused refactoring, but it should be done next step after this PR

## 📑 Test Plan

- All existing tests work
- Volatile expression works with `<let/>`

## ⏭ Next Steps

Remove AST `.connect()`

### CODE REVIEW

File: packages/runtime-html/src/binding/attribute.ts [hunk 1]
```diff
@@ -122,29 +122,29 @@ export class AttributeBinding implements IPartialConnectableBinding {
...
+        }
       }
 
       if (newValue !== oldValue) {
         if (shouldQueueFlush) {
           flags |= LifecycleFlags.noTargetObserverQueue;
           this.task?.cancel();
-          targetObserver.task?.cancel();
-          targetObserver.task = this.task = this.$scheduler.queueRenderTask(() => {
+          this.task = this.$scheduler.queueRenderTask(() => {
```
Comment (bigopon) [inline packages/runtime-html/src/binding/attribute.ts:139]: For all built in observers, they are created per binding, so there's no need to stamp the task on the observer/accessor itself.
In a future refactoring, we should be able to remove all per binding queue and centralize them like in #944 
So please ignore this change.

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://55407-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://55461-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://55472-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://55479-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://55580-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Comment by AureliaEffect** [comment]
JS Framework Benchmark: https://55587-121594173-gh.circle-artifacts.com/0/test/js-framework-benchmark/webdriver-ts-results/table.html

**Pull request merged**